pipeline/flows/notifications.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+from typing import Dict, List
+
+import prefect
+from prefect import Flow, Parameter
+from prefect.tasks.secrets import PrefectSecret
+
+from pipeline.extract import extract_users
+from pipeline.notifications import NewMessageNotifier
+from pipeline.transform import convert_users_from_df
+from pipeline.types import ChatData, NotificationInfo, NotificationInput
+from pipeline.utils.firebase import initialize_firebase_for_prefect
+
+
+@prefect.task
+def get_users_to_notify(inp: NotificationInput) -> List[NotificationInfo]:
+    logger = prefect.context.get("logger")
+    # add other types of notifiers once they are implemented
+    notifier = NewMessageNotifier()
+    logger.info("Generating notifications")
+    # refactor to account for additional notification types once they're implemented
+    notif_info_lst = notifier.get_notifications(inp=inp)
+    logger.info(
+        f"Done. Number of notifications generated: {len(notif_info_lst)}"
+    )
+    return notif_info_lst
+
+
+def notif_hierarchy(notifs: List[NotificationInfo]) -> List[NotificationInfo]:
```

Comment:
What do you think about how I went about implementing the hierarchy? Do you agree with what I've decided the hierarchy should be? Do you believe I should do something different? I'm open to any suggestions to improve the hierarchy function in any way.

Reply:
This looks fine! Let's leave it how it is because I don't want you to focus on this one too much.

@azuhdz7 is working on their own implementation of this, what you have works for now, but they have another approach and tests that we can have you review.

Reply:
I see. Excellent. I'm interested to see @azuhdz7 implementation. I'm sure it'll be more effective than I have at the moment.

Reply:
Awesome, we will do that in a follow-up PR after #366 merges!